Extensions/RosettaTraining/Scripts/TensorFlow/model.py

```diff
@@ -0,0 +1,246 @@
+    return card_id_embed
+
+  def _model_current_hand_card(self, input_getter):
+    outputs = []
+
+    card_id = input_getter.get_next_slice(1)
+    card_id_embed = self._get_embedded_hand_card_id(card_id)
+    if card_id_embed is not None:
+      outputs.append(card_id_embed)
+
+    card_features = input_getter.get_next_slice(
+        data_reader.kCurrentHandCardFeatures)
+    card1 = tf.layers.dense(
+        name='hand_card_1',
+        inputs=card_features,
+        units=self.kHandConvolutionHidden1,
+        reuse=tf.AUTO_REUSE,
+        activation=tf.nn.relu)
+    outputs.append(card1)
+
+    return outputs
+
+  def _model_current_hand(self, input_getter):
+    outputs = []
+    outputs.append(
+        input_getter.get_next_slice(data_reader.kCurrentHandFeatures))
+
+    for _ in range(data_reader.kCurrentHandCards):
+      outputs.extend(self._model_current_hand_card(input_getter))
+
+    return outputs
+
+  @staticmethod
+  def _model_board_features(input_getter):
+    return [input_getter.get_rest()]
+
+  @staticmethod
+  def _residual_block_unit(scope, inputs, hidden, add_input=None):
+    with tf.variable_scope(scope):
+      output = tf.contrib.layers.fully_connected(
+          inputs,
+          hidden,
+          activation_fn=None,
+          scope='dense')
+
+      if add_input is not None:
+        output = tf.add(output, add_input)
+
+      output = tf.nn.relu(output, 'relu')
+
+      return output
+
+  def _residual_block(self, idx, inputs, hidden):
+    name = 'residual_' + str(idx)
+
+    dense1 = self._residual_block_unit(
+        name + '_1',
+        inputs,
+        hidden)
+
+    dense2 = self._residual_block_unit(
+        name + '_2',
+        dense1,
+        hidden,
+        add_input=inputs)
+
+    return dense2
+
+  def set_mode(self, mode):
+    self._mode = mode
+
+  def get_model(self, features, labels):
+    feature = features["x"]
+    feature = tf.to_float(feature)
+
+    input_getter = NextInputGetter(feature)
+
+    inputs = []
+
+    inputs.extend(self._model_hero(input_getter))  # current hero
+    inputs.extend(self._model_hero(input_getter))  # opponent hero
+    inputs.extend(self._model_minions(input_getter))  # current minions
+    inputs.extend(self._model_minions(input_getter))  # opponent minions
+    inputs.extend(self._model_current_hand(input_getter))  # current hand
+    inputs.extend(self._model_board_features(input_getter))  # board features
+
+    inputs = tf.concat(inputs, 1)
+
+    dense1 = tf.layers.dense(
+        name='dense1',
+        inputs=inputs,
+        units=self.kResidualBlockFeatures,
+        activation=tf.nn.relu)
+
+    prev = dense1
+    for i in range(0, self.kResidualBlocks):
+      prev = self._residual_block(i+1, prev, self.kResidualBlockFeatures)
+
+    final = tf.layers.dense(
+        name='final',
+        inputs=prev,
+        units=1,
+        activation=None)
+		
+    final_argmax = tf.argmax(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [local variable 'final_argmax' is assigned to but never used](https://app.codacy.com/manual/utilForever/RosettaStone/pullRequest?prid=4748929)